orca-run-multiple-pipelines/src/main/java/io/armory/plugin/smp/execution/MyExecutionLauncher.java

```diff
@@ -285,26 +287,96 @@ private void persistExecution(PipelineExecution execution) {
         executionRepository.store(execution);
     }
 
-    private final boolean getBoolean(Map<String, ?> map, String key) {
+    private static boolean getBoolean(Map<String, ?> map, String key) {
         return parseBoolean(getString(map, key));
     }
 
-    private final String getString(Map<String, ?> map, String key) {
+    private static int getInt(Map<String, ?> map, String key) {
+        return map.containsKey(key) ? parseInt(getString(map, key)) : 0;
+    }
+
+    private static String getString(Map<String, ?> map, String key) {
         return map.containsKey(key) ? map.get(key).toString() : null;
     }
 
-    private final <K, V> Map<K, V> getMap(Map<String, ?> map, String key) {
+    private static <K, V> Map<K, V> getMap(Map<String, ?> map, String key) {
         Map<K, V> result = (Map<K, V>) map.get(key);
         return result == null ? emptyMap() : result;
     }
 
-    private final List<Map<String, Object>> getList(Map<String, ?> map, String key) {
+    private static List<Map<String, Object>> getList(Map<String, ?> map, String key) {
         List<Map<String, Object>> result = (List<Map<String, Object>>) map.get(key);
         return result == null ? emptyList() : result;
     }
 
-    private final <E extends Enum<E>> E getEnum(Map<String, ?> map, String key, Class<E> type) {
+    private static <E extends Enum<E>> E getEnum(Map<String, ?> map, String key, Class<E> type) {
         String value = (String) map.get(key);
         return value != null ? Enum.valueOf(type, value) : null;
     }
+
+    /**
+     * for security reasons, we only want to allow a few explicitly configured ad-hoc operations.
+     * Every other operation should be blocked. Depending on how the allowed ad-hoc operations are
+     * configured, we also check to see if the user performing this action is allowed to run this
+     * operation
+     *
+     * @param execution pipeline execution
+     * @throws AccessDeniedException if the orchestration action is explicitly disabled or if the user
+     *     is not allowed to perform that action
+     */
+    private void checkIfPipelineCanBeStarted(PipelineExecution execution) {
```

Comment:
Is it possible to test this new behaviour with unit tests? If so, let's test it!

Reply:
Hi Chris,

This code has been replicated from the parent class without any changes on our end. I can copy the unit tests from the Orca package to the plugin, but it might take some time to complete.

https://github.com/spinnaker/orca/blob/master/orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/PipelineExecutionLauncherSpec.groovy